### PR TITLE
Wrong diff in case of dictionary has dotted keys in ignore in dictdiffer.

### PR DESCRIPTION
**Pull request by yoyonel**
Resolve issue #107.

Use a custom type `DottedIgnoreKey` to deactivate interpretation (interpolation) of keys (with dot inside).
Add unit-tests to check the correctness/fix.

### CODE REVIEW

File: dictdiffer/utils.py [hunk 1]
```diff
@@ -16,6 +16,15 @@
 EPSILON = sys.float_info.epsilon
 
 
+class DottedIgnoreKey(str):
+    """Custom type to specify dotted ignore key.
+
+    This custom type help to desactivate dotted interpretation.
```
Comment (jirikuncar) [inline dictdiffer/utils.py]:
```suggestion
    This custom type helps to deactivate dotted interpretation.
```

File: dictdiffer/utils.py [hunk 2]
```diff
@@ -16,6 +16,15 @@
...
+    """
+
+    pass
```
Comment (jirikuncar) [inline dictdiffer/utils.py]: It's not necessary when the `class` has a docstring.
 
```suggestion
    pass
```

Comment (yoyonel) [reply dictdiffer/utils.py]: Oki good to know, thx :)

**Comment by jirikuncar** [comment]
@yoyonel can you try your testcase with `diff(..., dot_notation=False)`

---
see https://github.com/inveniosoftware/dictdiffer/pull/123

**Comment by yoyonel** [comment]
I've rebased and add this option in my testcases.

**Comment by jirikuncar** [comment]
@yoyonel thank you for your time! I have combined #123 and your tests into a simpler solution.

**Comment by yoyonel** [comment]
You're welcome. Nice pytest refactoring !

**Pull request merged**